pkg/report: panic: non matching oops

We are getting these panic on managers:

```
panic: non matching oops

goroutine 37717 [running]:
github.com/google/syzkaller/pkg/report.extractDescription(0x0, 0x0, 0x0, 0x13bcee0, 0x0, 0x0, 0x0, 0xffffffffffffffff, 0x8, 0x0, ...)
        /syzkaller/gopath/src/github.com/google/syzkaller/pkg/report/report.go:153 +0x6de
github.com/google/syzkaller/pkg/report.(*linux).Parse(0xc43195d0e0, 0xc43b31dc00, 0xaf4, 0xaf4, 0xc4310e43e8)
        /syzkaller/gopath/src/github.com/google/syzkaller/pkg/report/linux.go:159 +0xf19
github.com/google/syzkaller/pkg/repro.Run(0xc43b31dc00, 0xaf4, 0xaf4, 0xc4201c8a00, 0x1164b80, 0xc43195d0e0, 0xc4206e1100, 0xc435f12f60, 0x4, 0x4, ...)
        /syzkaller/gopath/src/github.com/google/syzkaller/pkg/repro/repro.go:75 +0x132
main.(*Manager).vmLoop.func2(0xc42d6c1dc0, 0xc4213bad00, 0xc435f12f60, 0x4, 0x4, 0xc420528240)
        /syzkaller/gopath/src/github.com/google/syzkaller/syz-manager/manager.go:429 +0xbf
created by main.(*Manager).vmLoop
        /syzkaller/gopath/src/github.com/google/syzkaller/syz-manager/manager.go:428 +0xbc8
```

--- a/pkg/report/freebsd.go
+++ b/pkg/report/freebsd.go
@@ -71,7 +71,7 @@ func (ctx *freebsd) Parse(output []byte) *Report {
 	if oops == nil {
 		return nil
 	}
-	rep.Title, _, _ = extractDescription(output[rep.StartPos:], oops)
+	rep.Title, _ = extractDescription(output[rep.StartPos:], oops)
 	return rep
 }
 

--- a/pkg/report/linux.go
+++ b/pkg/report/linux.go
@@ -84,7 +84,9 @@ func (ctx *linux) Parse(output []byte) *Report {
 		Output: output,
 	}
 	var oops *oops
-	var textPrefix [][]byte
+	var logReportPrefix [][]byte
+	var consoleReportPrefix [][]byte
+	var consoleReport []byte
 	textLines := 0
 	skipText := false
 	for pos := 0; pos < len(output); {
@@ -106,6 +108,12 @@ func (ctx *linux) Parse(output []byte) *Report {
 			}
 			rep.EndPos = next
 		}
+		if oops == nil {
+			logReportPrefix = append(logReportPrefix, append([]byte{}, output[pos:next]...))
+			if len(logReportPrefix) > 5 {
+				logReportPrefix = logReportPrefix[1:]
+			}
+		}
 		if ctx.consoleOutputRe.Match(output[pos:next]) &&
 			(!ctx.questionableRe.Match(output[pos:next]) ||
 				bytes.Index(output[pos:next], ctx.eoi) != -1) {
@@ -115,18 +123,11 @@ func (ctx *linux) Parse(output []byte) *Report {
 				lineEnd--
 			}
 			if oops == nil {
-				textPrefix = append(textPrefix, append([]byte{}, output[lineStart:lineEnd]...))
-				if len(textPrefix) > 5 {
-					textPrefix = textPrefix[1:]
+				consoleReportPrefix = append(consoleReportPrefix, append([]byte{}, output[lineStart:lineEnd]...))
+				if len(consoleReportPrefix) > 5 {
+					consoleReportPrefix = consoleReportPrefix[1:]
 				}
 			} else {
-				// Prepend 5 lines preceding start of the report,
-				// they can contain additional info related to the report.
-				for _, prefix := range textPrefix {
-					rep.Report = append(rep.Report, prefix...)
-					rep.Report = append(rep.Report, '\n')
-				}
-				textPrefix = nil
 				textLines++
 				ln := output[lineStart:lineEnd]
 				skipLine := skipText
@@ -146,8 +147,8 @@ func (ctx *linux) Parse(output []byte) *Report {
 					skipLine = true
 				}
 				if !skipLine {
-					rep.Report = append(rep.Report, ln...)
-					rep.Report = append(rep.Report, '\n')
+					consoleReport = append(consoleReport, ln...)
+					consoleReport = append(consoleReport, '\n')
 				}
 			}
 		}
@@ -156,11 +157,31 @@ func (ctx *linux) Parse(output []byte) *Report {
 	if oops == nil {
 		return nil
 	}
-	title, report, format := extractDescription(ctx.extractConsoleOutput(output[rep.StartPos:]), oops)
-	if title == "" {
-		title, report, format = extractDescription(output[rep.StartPos:], oops)
+	var report []byte
+	var reportPrefix [][]byte
+	// Try extracting report from console output only.
+	title, format := extractDescription(consoleReport, oops)
+	if len(title) != 0 {
+		// Success.
+		report = consoleReport
+		reportPrefix = consoleReportPrefix
+	} else {
+		// Failure. Try extracting report from the whole log.
+		report = output[rep.StartPos:]
+		reportPrefix = logReportPrefix
+		title, format = extractDescription(report, oops)
+		if len(title) == 0 {
+			panic(fmt.Sprintf("non matching oops for %q in:\n%s", oops.header, report))
+		}
 	}
 	rep.Title = title
+	// Prepend 5 lines preceding start of the report,
+	// they can contain additional info related to the report.
+	for _, prefix := range reportPrefix {
+		rep.Report = append(rep.Report, prefix...)
+		rep.Report = append(rep.Report, '\n')
+	}
+	rep.Report = append(rep.Report, report...)
 	rep.Corrupted = ctx.isCorrupted(title, report, format)
 	// Executor PIDs are not interesting.
 	rep.Title = executorRe.ReplaceAllLiteralString(rep.Title, "syz-executor")
@@ -176,10 +197,6 @@ func (ctx *linux) Parse(output []byte) *Report {
 	rep.Title = funcRe.ReplaceAllString(rep.Title, "$1")
 	// CPU numbers are not interesting.
 	rep.Title = cpuRe.ReplaceAllLiteralString(rep.Title, "CPU")
-	// TODO: broken: https://github.com/google/syzkaller/issues/457
-	if len(rep.Report) == 0 {
-		rep.Report = []byte("NO REPORT")
-	}
 	return rep
 }
 
@@ -283,30 +300,6 @@ func symbolizeLine(symbFunc func(bin string, pc uint64) ([]symbolizer.Frame, err
 		symbolized = append(symbolized, modified...)
 	}
 	return symbolized
-}
-
-func (ctx *linux) extractConsoleOutput(output []byte) (result []byte) {
-	for pos := 0; pos < len(output); {
-		next := bytes.IndexByte(output[pos:], '\n')
-		if next != -1 {
-			next += pos
-		} else {
-			next = len(output)
-		}
-		if ctx.consoleOutputRe.Match(output[pos:next]) &&
-			(!ctx.questionableRe.Match(output[pos:next]) ||
-				bytes.Index(output[pos:next], ctx.eoi) != -1) {
-			lineStart := bytes.Index(output[pos:next], []byte("] ")) + pos + 2
-			lineEnd := next
-			if lineEnd != 0 && output[lineEnd-1] == '\r' {
-				lineEnd--
-			}
-			result = append(result, output[lineStart:lineEnd]...)
-			result = append(result, '\n')
-		}
-		pos = next + 1
-	}
-	return
 }
 
 func (ctx *linux) extractGuiltyFile(report []byte) string {
@@ -549,6 +542,11 @@ var linuxOopses = []*oops{
 				title: compile("BUG: using __this_cpu_add\\(\\) in preemptible (.*)"),
 				fmt:   "BUG: using __this_cpu_add() in preemptible %[1]v",
 			},
+			{
+				title:        compile("BUG: executor-detected bug"),
+				fmt:          "BUG: executor-detected bug",
+				noStackTrace: true,
+			},
 		},
 		[]*regexp.Regexp{
 			// Android prints this sometimes during boot.
@@ -662,39 +660,19 @@ var linuxOopses = []*oops{
 				fmt:   "inconsistent lock state in %[1]v",
 			},
 			{
-				title: compile("INFO: rcu_preempt detected stalls(?:.*\\n)+?.*</IRQ>.*\n(?:.*rcu.*\\n)+? {{FUNC}}"),
+				title: compile("INFO: rcu_(?:preempt|sched|bh) detected(?: expedited)? stalls(?:.*\\n)+?.*</IRQ>.*\n(?:.*rcu.*\\n)+? {{FUNC}}"),
 				fmt:   "INFO: rcu detected stall in %[1]v",
 			},
 			{
-				title: compile("INFO: rcu_preempt detected stalls"),
+				title: compile("INFO: rcu_(?:preempt|sched|bh) detected(?: expedited)? stalls"),
 				fmt:   "INFO: rcu detected stall",
 			},
 			{
-				title: compile("INFO: rcu_sched detected(?: expedited)? stalls(?:.*\\n)+?.*</IRQ>.*\n(?:.*rcu.*\\n)+? {{FUNC}}"),
+				title: compile("INFO: rcu_(?:preempt|sched|bh) self-detected stall on CPU(?:.*\\n)+?.*</IRQ>.*\n(?:.*rcu.*\\n)+? {{FUNC}}"),
 				fmt:   "INFO: rcu detected stall in %[1]v",
 			},
 			{
-				title: compile("INFO: rcu_sched detected(?: expedited)? stalls"),
-				fmt:   "INFO: rcu detected stall",
-			},
-			{
-				title: compile("INFO: rcu_preempt self-detected stall on CPU(?:.*\\n)+?.*</IRQ>.*\n(?:.*rcu.*\\n)+? {{FUNC}}"),
-				fmt:   "INFO: rcu detected stall in %[1]v",
-			},
-			{
-				title: compile("INFO: rcu_preempt self-detected stall on CPU"),
-				fmt:   "INFO: rcu detected stall",
-			},
-			{
-				title: compile("INFO: rcu_sched self-detected stall on CPU(?:.*\\n)+?.*</IRQ>.*\n(?:.*rcu.*\\n)+? {{FUNC}}"),
-				fmt:   "INFO: rcu detected stall in %[1]v",
-			},
-			{
-				title: compile("INFO: rcu_sched self-detected stall on CPU"),
-				fmt:   "INFO: rcu detected stall",
-			},
-			{
-				title: compile("INFO: rcu_bh detected stalls on CPU"),
+				title: compile("INFO: rcu_(?:preempt|sched|bh) self-detected stall on CPU"),
 				fmt:   "INFO: rcu detected stall",
 			},
 			{

--- a/pkg/report/linux_test.go
+++ b/pkg/report/linux_test.go
@@ -633,59 +633,104 @@ r0 = ioctl$KVM_CREATE_VM(0xffffffffffffffff, 0xae01, 0x0)
 [ 1722.511384] BUG: sleeping function called from invalid context at include/linux/wait.h:1095 
 [ 1722.511384] in_atomic(): 1, irqs_disabled(): 0, pid: 3658, name: syz-fuzzer 
 `, `BUG: sleeping function called from invalid context at include/linux/wait.h:LINE `, true,
-		},
-
-		// TODO: broken: https://github.com/google/syzkaller/issues/457
-		/* {
-					`
-		[  277.780013] INFO: rcu_sched self-detected stall on CPU
-		[  277.781045] INFO: rcu_sched detected stalls on CPUs/tasks:
-		[  277.781153] 	1-...: (65000 ticks this GP) idle=395/140000000000001/0 softirq=122875/122875 fqs=16248
-		[  277.781197] 	(detected by 0, t=65002 jiffies, g=72940, c=72939, q=1777)
-		[  277.781212] Sending NMI from CPU 0 to CPUs 1:
-		[  277.782014] NMI backtrace for cpu 1
-		[  277.782014] CPU: 1 PID: 12579 Comm: syz-executor0 Not tainted 4.11.0-rc3+ #71
-		[  277.782014] Hardware name: Google Google Compute Engine/Google Compute Engine, BIOS Google 01/01/2011
-		[  277.782014] task: ffff8801d379e140 task.stack: ffff8801cd590000
-		[  277.782014] RIP: 0010:io_serial_in+0x6b/0x90
-		[  277.782014] RSP: 0018:ffff8801dbf066a0 EFLAGS: 00000002
-		[  277.782014] RAX: dffffc0000000000 RBX: 00000000000003fd RCX: 0000000000000000
-		[  277.782014] RDX: 00000000000003fd RSI: 0000000000000005 RDI: ffffffff87020018
-		[  277.782014] RBP: ffff8801dbf066b0 R08: 0000000000000003 R09: 0000000000000001
-		[  277.782014] R10: dffffc0000000000 R11: ffffffff867ba200 R12: ffffffff8701ffe0
-		[  277.782014] R13: 0000000000000020 R14: fffffbfff0e04041 R15: fffffbfff0e04005
-		[  277.782014] FS:  00007fce6fc10700(0000) GS:ffff8801dbf00000(0000) knlGS:0000000000000000
-		[  277.782014] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
-		[  277.782014] CR2: 000000002084fffc CR3: 00000001c4500000 CR4: 00000000001406e0
-		[  277.782014] Call Trace:
-		[  277.782014]  <IRQ>
-		[  277.782014]  wait_for_xmitr+0x89/0x1c0
-		[  277.782014]  ? wait_for_xmitr+0x1c0/0x1c0
-		[  277.782014]  serial8250_console_putchar+0x1f/0x60
-		[  277.782014]  uart_console_write+0x57/0xe0
-		[  277.782014]  serial8250_console_write+0x423/0x840
-		[  277.782014]  ? check_noncircular+0x20/0x20
-		[  277.782014]  hrtimer_interrupt+0x1c2/0x5e0
-		[  277.782014]  local_apic_timer_interrupt+0x6f/0xe0
-		[  277.782014]  smp_apic_timer_interrupt+0x71/0xa0
-		[  277.782014]  apic_timer_interrupt+0x93/0xa0
-		[  277.782014] RIP: 0010:debug_lockdep_rcu_enabled.part.19+0xf/0x60
-		[  277.782014] RSP: 0018:ffff8801cd596778 EFLAGS: 00000202 ORIG_RAX: ffffffffffffff10
-		[  277.782014] RAX: dffffc0000000000 RBX: 1ffff10039ab2cf7 RCX: ffffc90001758000
-		[  277.782014] RDX: 0000000000000004 RSI: ffffffff840561f1 RDI: ffffffff852a75c0
-		[  277.782014] RBP: ffff8801cd596780 R08: 0000000000000001 R09: 0000000000000000
-		[  277.782014] R10: dffffc0000000000 R11: ffffffff867ba200 R12: 1ffff10039ab2d1b
-		[  277.782014] R13: ffff8801c44d1880 R14: ffff8801cd596918 R15: ffff8801d9b47840
-		[  277.782014]  </IRQ>
-		[  277.782014]  ? __write_space+0x5b1/0x920
-		[  277.782014]  debug_lockdep_rcu_enabled+0x77/0x90
-		[  292.742848] sctp: [Deprecated]: syz-executor0 (pid 22154) Use of int in max_burst socket option deprecated.
-		[  277.782014]  __sctp_write_space+0x5b6/0x920
-		[  277.782014]  ? __sctp_write_space+0x3f7/0x920
-		[  277.782014]  ? sctp_transport_lookup_process+0x190/0x190
-		[  277.782014]  ? trace_hardirqs_on_thunk+0x1a/0x1c
-		`, `INFO: rcu detected stall in __sctp_write_space`, false,
-				},*/{
+		}, {
+			`
+[  277.780013] INFO: rcu_sched self-detected stall on CPU
+[  277.781045] INFO: rcu_sched detected stalls on CPUs/tasks:
+[  277.781153] 	1-...: (65000 ticks this GP) idle=395/140000000000001/0 softirq=122875/122875 fqs=16248
+[  277.781197] 	(detected by 0, t=65002 jiffies, g=72940, c=72939, q=1777)
+[  277.781212] Sending NMI from CPU 0 to CPUs 1:
+[  277.782014] NMI backtrace for cpu 1
+[  277.782014] CPU: 1 PID: 12579 Comm: syz-executor0 Not tainted 4.11.0-rc3+ #71
+[  277.782014] Hardware name: Google Google Compute Engine/Google Compute Engine, BIOS Google 01/01/2011
+[  277.782014] task: ffff8801d379e140 task.stack: ffff8801cd590000
+[  277.782014] RIP: 0010:io_serial_in+0x6b/0x90
+[  277.782014] RSP: 0018:ffff8801dbf066a0 EFLAGS: 00000002
+[  277.782014] RAX: dffffc0000000000 RBX: 00000000000003fd RCX: 0000000000000000
+[  277.782014] RDX: 00000000000003fd RSI: 0000000000000005 RDI: ffffffff87020018
+[  277.782014] RBP: ffff8801dbf066b0 R08: 0000000000000003 R09: 0000000000000001
+[  277.782014] R10: dffffc0000000000 R11: ffffffff867ba200 R12: ffffffff8701ffe0
+[  277.782014] R13: 0000000000000020 R14: fffffbfff0e04041 R15: fffffbfff0e04005
+[  277.782014] FS:  00007fce6fc10700(0000) GS:ffff8801dbf00000(0000) knlGS:0000000000000000
+[  277.782014] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
+[  277.782014] CR2: 000000002084fffc CR3: 00000001c4500000 CR4: 00000000001406e0
+[  277.782014] Call Trace:
+[  277.782014]  <IRQ>
+[  277.782014]  wait_for_xmitr+0x89/0x1c0
+[  277.782014]  ? wait_for_xmitr+0x1c0/0x1c0
+[  277.782014]  serial8250_console_putchar+0x1f/0x60
+[  277.782014]  uart_console_write+0x57/0xe0
+[  277.782014]  serial8250_console_write+0x423/0x840
+[  277.782014]  ? check_noncircular+0x20/0x20
+[  277.782014]  hrtimer_interrupt+0x1c2/0x5e0
+[  277.782014]  local_apic_timer_interrupt+0x6f/0xe0
+[  277.782014]  smp_apic_timer_interrupt+0x71/0xa0
+[  277.782014]  apic_timer_interrupt+0x93/0xa0
+[  277.782014] RIP: 0010:debug_lockdep_rcu_enabled.part.19+0xf/0x60
+[  277.782014] RSP: 0018:ffff8801cd596778 EFLAGS: 00000202 ORIG_RAX: ffffffffffffff10
+[  277.782014] RAX: dffffc0000000000 RBX: 1ffff10039ab2cf7 RCX: ffffc90001758000
+[  277.782014] RDX: 0000000000000004 RSI: ffffffff840561f1 RDI: ffffffff852a75c0
+[  277.782014] RBP: ffff8801cd596780 R08: 0000000000000001 R09: 0000000000000000
+[  277.782014] R10: dffffc0000000000 R11: ffffffff867ba200 R12: 1ffff10039ab2d1b
+[  277.782014] R13: ffff8801c44d1880 R14: ffff8801cd596918 R15: ffff8801d9b47840
+[  277.782014]  </IRQ>
+[  277.782014]  ? __write_space+0x5b1/0x920
+[  277.782014]  debug_lockdep_rcu_enabled+0x77/0x90
+[  277.782014]  __sctp_write_space+0x5b6/0x920
+[  277.782014]  ? __sctp_write_space+0x3f7/0x920
+[  277.782014]  ? sctp_transport_lookup_process+0x190/0x190
+[  277.782014]  ? trace_hardirqs_on_thunk+0x1a/0x1c
+`, `INFO: rcu detected stall in __sctp_write_space`, false,
+		}, {
+			`
+[  277.780013] INFO: rcu_sched self-detected stall on CPU
+[  277.781045] INFO: rcu_sched detected stalls on CPUs/tasks:
+[  277.781153] 	1-...: (65000 ticks this GP) idle=395/140000000000001/0 softirq=122875/122875 fqs=16248
+[  277.781197] 	(detected by 0, t=65002 jiffies, g=72940, c=72939, q=1777)
+[  277.781212] Sending NMI from CPU 0 to CPUs 1:
+[  277.782014] NMI backtrace for cpu 1
+[  277.782014] CPU: 1 PID: 12579 Comm: syz-executor0 Not tainted 4.11.0-rc3+ #71
+[  277.782014] Hardware name: Google Google Compute Engine/Google Compute Engine, BIOS Google 01/01/2011
+[  277.782014] task: ffff8801d379e140 task.stack: ffff8801cd590000
+[  277.782014] RIP: 0010:io_serial_in+0x6b/0x90
+[  277.782014] RSP: 0018:ffff8801dbf066a0 EFLAGS: 00000002
+[  277.782014] RAX: dffffc0000000000 RBX: 00000000000003fd RCX: 0000000000000000
+[  277.782014] RDX: 00000000000003fd RSI: 0000000000000005 RDI: ffffffff87020018
+[  277.782014] RBP: ffff8801dbf066b0 R08: 0000000000000003 R09: 0000000000000001
+[  277.782014] R10: dffffc0000000000 R11: ffffffff867ba200 R12: ffffffff8701ffe0
+[  277.782014] R13: 0000000000000020 R14: fffffbfff0e04041 R15: fffffbfff0e04005
+[  277.782014] FS:  00007fce6fc10700(0000) GS:ffff8801dbf00000(0000) knlGS:0000000000000000
+[  277.782014] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
+[  277.782014] CR2: 000000002084fffc CR3: 00000001c4500000 CR4: 00000000001406e0
+[  277.782014] Call Trace:
+[  277.782014]  <IRQ>
+[  277.782014]  wait_for_xmitr+0x89/0x1c0
+[  277.782014]  ? wait_for_xmitr+0x1c0/0x1c0
+[  277.782014]  serial8250_console_putchar+0x1f/0x60
+[  277.782014]  uart_console_write+0x57/0xe0
+[  277.782014]  serial8250_console_write+0x423/0x840
+[  277.782014]  ? check_noncircular+0x20/0x20
+[  277.782014]  hrtimer_interrupt+0x1c2/0x5e0
+[  277.782014]  local_apic_timer_interrupt+0x6f/0xe0
+[  277.782014]  smp_apic_timer_interrupt+0x71/0xa0
+[  277.782014]  apic_timer_interrupt+0x93/0xa0
+[  277.782014] RIP: 0010:debug_lockdep_rcu_enabled.part.19+0xf/0x60
+[  277.782014] RSP: 0018:ffff8801cd596778 EFLAGS: 00000202 ORIG_RAX: ffffffffffffff10
+[  277.782014] RAX: dffffc0000000000 RBX: 1ffff10039ab2cf7 RCX: ffffc90001758000
+[  277.782014] RDX: 0000000000000004 RSI: ffffffff840561f1 RDI: ffffffff852a75c0
+[  277.782014] RBP: ffff8801cd596780 R08: 0000000000000001 R09: 0000000000000000
+[  277.782014] R10: dffffc0000000000 R11: ffffffff867ba200 R12: 1ffff10039ab2d1b
+[  277.782014] R13: ffff8801c44d1880 R14: ffff8801cd596918 R15: ffff8801d9b47840
+[  277.782014]  </IRQ>
+[  277.782014]  ? __write_space+0x5b1/0x920
+[  277.782014]  debug_lockdep_rcu_enabled+0x77/0x90
+[  292.742848] sctp: [Deprecated]: syz-executor0 (pid 22154) Use of int in max_burst socket option deprecated.
+[  277.782014]  __sctp_write_space+0x5b6/0x920
+[  277.782014]  ? __sctp_write_space+0x3f7/0x920
+[  277.782014]  ? sctp_transport_lookup_process+0x190/0x190
+[  277.782014]  ? trace_hardirqs_on_thunk+0x1a/0x1c
+`, `INFO: rcu detected stall`, false,
+		}, {
 			`
 [ 1722.511384] INFO: rcu_preempt detected stalls on CPUs/tasks: { 2} (detected by 0, t=65008 jiffies, g=48068, c=48067, q=7339)
 `, `INFO: rcu detected stall`, true,
@@ -1734,8 +1779,7 @@ Booting the kernel.
 [    1.971945] Kernel memory protection disabled.
 `, `unexpected kernel reboot`, false,
 		},
-		// TODO: broken: https://github.com/google/syzkaller/issues/457
-		//{`BUG: executor-detected bug`, `BUG: executor-detected bug`, false},
+		{`BUG: executor-detected bug`, `BUG: executor-detected bug`, false},
 		{`INFO:`, `INFO:`, true},
 	}
 	testParse(t, "linux", tests)

--- a/pkg/report/report.go
+++ b/pkg/report/report.go
@@ -128,7 +128,7 @@ func matchOops(line []byte, oops *oops, ignores []*regexp.Regexp) int {
 	return match
 }
 
-func extractDescription(output []byte, oops *oops) (desc string, report []byte, format oopsFormat) {
+func extractDescription(output []byte, oops *oops) (desc string, format oopsFormat) {
 	startPos := -1
 	for _, f := range oops.formats {
 		match := f.title.FindSubmatchIndex(output)
@@ -144,14 +144,11 @@ func extractDescription(output []byte, oops *oops) (desc string, report []byte, 
 			args = append(args, string(output[match[i]:match[i+1]]))
 		}
 		desc = fmt.Sprintf(f.fmt, args...)
-		report = output[startPos:]
 		format = f
 	}
 	if len(desc) == 0 {
 		pos := bytes.Index(output, oops.header)
 		if pos == -1 {
-			// TODO: broken: https://github.com/google/syzkaller/issues/457
-			// panic(fmt.Sprintf("non matching oops for %q in:\n%s", oops.header, output))
 			return
 		}
 		end := bytes.IndexByte(output[pos:], '\n')
@@ -161,7 +158,6 @@ func extractDescription(output []byte, oops *oops) (desc string, report []byte, 
 			end += pos
 		}
 		desc = string(output[pos:end])
-		report = output[pos:]
 	}
 	if len(desc) > 0 && desc[len(desc)-1] == '\r' {
 		desc = desc[:len(desc)-1]

--- a/vm/vm.go
+++ b/vm/vm.go
@@ -151,15 +151,17 @@ func MonitorExecution(outc <-chan []byte, errc <-chan error, reporter report.Rep
 		if rep == nil {
 			panic(fmt.Sprintf("reporter.ContainsCrash/Parse disagree:\n%s", output[matchPos:]))
 		}
-		start := rep.StartPos + matchPos - beforeContext
+		start := matchPos + rep.StartPos - beforeContext
 		if start < 0 {
 			start = 0
 		}
-		end := rep.EndPos + matchPos + afterContext
+		end := matchPos + rep.EndPos + afterContext
 		if end > len(output) {
 			end = len(output)
 		}
 		rep.Output = output[start:end]
+		rep.StartPos += matchPos - start
+		rep.EndPos += matchPos - start
 		return rep
 	}
 

